source/bao_hyp/config.rst

```diff
@@ -65,39 +65,404 @@ describe the image running on the VM (see `Guest Image`_);
+parameters such as img_name, base_addr, load_addr, and size. This practice not only fosters
+consistency but also enhances reliability in your VM setup. For instance:
+
+- **IMAGE_BUILTIN**: Simplifies system configuration by leveraging Bao's default knowledge of the
+  image location. No separate configuration or loading of guest images through a bootloader is
+  required, and adjustments to the size of guest images are unnecessary.
+
+- **IMAGE_LOADED**: Highly recommended, especially for MPU systems, where manual allocation of space
+  for the guest image can be challenging if embedded in Bao's binary. Without utilizing LOADED, Bao
+  may need to copy the image, potentially resulting in space wastage
+
+Moreover, if the ``separately_loaded`` parameter is configured as false, the hypervisor interprets
+this setting as the offset of the built-in guest image within its own image, denoted as
+``VM_IMAGE_OFFSET``. During run-time, the hypervisor adjusts this value to be subsequently
+interpreted as a physical address. This adjustment involves adding the address at which the
+hypervisor itself was loaded. However, if the ``separately_loaded`` parameter is configured as
+true, the guest image is not embedded in the hypervisor image; instead, it is loaded independently.
+For more details, refer to the figure below.
+
+.. figure:: img/guest-image.svg
+    :align: center
+    :name: vm-image-fig
+
 
 Virtual Machine Configuration
 *****************************
 
+The VM configuration enables users to define the characteristics of each virtualized platform. It
+encompasses critical details that define the VM's run-time environment, performance, and overall
+capabilities. The virtual machine configuration is performed by populating the structure `struct
+vm_platform`, outlined below:
+
+.. code-block:: c
+
+    struct vm_platform {
+        size_t cpu_num;
+        size_t region_num;
+        struct vm_mem_region* regions;
+        size_t ipc_num;
+        struct ipc* ipcs;
+        size_t dev_num;
+        struct vm_dev_region* devs;
+        bool mmu;
+        struct arch_vm_platform arch;
+    }
+
+By customizing this configuration, users can tailor the virtual platform to suit specific workload
+requirements and application needs for their virtual machines. The configuration encompasses the
+definition of:
+
+- **Number of CPUs** - see details in `Number of vCPUs`_;
+- **Memory regions** - see details in `Memory Regions`_;`
+- **Inter-Process Comunication (IPC)** - see details in `Inter-Process Communication (IPC)`_;
+- **Devices** - see details in `Devices`_;
+- **Memory Management** - see details in `Memory Management`_;
+- **Architectural-Specific Configurations** - see details in `Architectural-Specific
+  Configurations`_;
+
 1. Number of vCPUs
 ##################
+.. _Number of vCPUs:
+
+- **cpu_num** [mandatory] - defines the number of CPUs assigned to the VM;
```

Comment:
Use inline format for the field name. Normalize on the overall document.

Reply:
Thanks, fixed!